src/app/clients/add/formComponents.tsx

```diff
@@ -0,0 +1,534 @@
+    padding-bottom: 1em;
+`;
+
+const Heading = styled.h1`
+    padding-bottom: 1em;
+`;
+
+const RequiredAsterisk = styled.span`
+    color: ${(props) => props.theme.errorColor};
+
+    &:before {
+        content: "*";
+    }
+`;
+
+const ErrorText = styled(Text)`
+    color: ${(props) => props.theme.errorColor};
+    margin-bottom: 3em;
+    text-align: center;
+`;
+
+const StyledButton = styled.button`
+    text-align: center;
+    width: 150px;
+    height: 40px;
+    border-radius: 10px;
+    border: solid 0 ${(props) => props.theme.foregroundColor};
+    background-color: ${(props) => props.theme.primaryBackgroundColor};
+    color: ${(props) => props.theme.primaryForegroundColor};
+
+    &:hover {
+        background-color: ${(props) => props.theme.secondaryBackgroundColor};
+        color: ${(props) => props.theme.secondaryForegroundColor};
+    }
+`;
+
+const genericFormCard = (
+    title: string,
+    required: boolean,
+    childComponent: React.ReactNode,
+    text?: string
+): React.ReactNode => {
+    return (
+        <>
+            <StyledCard>
+                <Subheading>
+                    {title} {required && <RequiredAsterisk />}
+                </Subheading>
+                {text && <Text>{text}</Text>}
+                {childComponent}
+            </StyledCard>
+        </>
+    );
+};
+
+const initialFields: Fields = {
+    fullName: "",
+    phoneNumber: "",
+    addressLine1: "",
+    addressLine2: "",
+    addressTown: "",
+    addressCounty: "",
+    addressPostcode: "",
+    adults: [
+        { personType: "adult", quantity: 0 },
+        { personType: "male", quantity: 0 },
+        { personType: "female", quantity: 0 },
+    ],
+    numberChildren: 0,
+    children: [],
+    dietaryRequirements: {},
+    feminineProducts: {},
+    babyProducts: null,
+    nappySize: "",
+    petFood: {},
+    otherItems: {},
+    deliveryInstructions: "",
+    extraInformation: "",
+};
+
+const initialFormErrors: FormErrors = {
+    fullName: Error.initial,
+    phoneNumber: Error.none,
+    addressLine1: Error.initial,
+    addressPostcode: Error.initial,
+    adults: Error.initial,
+    numberChildren: Error.initial,
+    nappySize: Error.none,
+};
+
+const AddClientForm: React.FC = () => {
+    const router = useRouter();
+    const [fields, setFields] = useState<Fields>(initialFields);
+    const [formErrors, setFormErrors] = useState<FormErrors>(initialFormErrors);
+    const [submitError, setSubmitError] = useState<Error>(Error.none);
+
+    useEffect(() => {
+        const childrenCopy: Person[] = [];
+        for (let index = 0; index < fields.numberChildren; index++) {
+            if (index < fields.children.length) {
+                childrenCopy.push(fields.children[index]);
+            } else {
+                childrenCopy.push({ personType: "child", age: null, quantity: 1 });
+            }
+        }
+        fieldSetter("children", childrenCopy);
+    }, [fields.numberChildren]); // eslint-disable-line
```

Comment:
can the eslint specify which rule we are disabling (something about exhaustive dependencies?)